lib/modules/manager/gomod/artifacts.ts

```diff
@@ -410,3 +413,18 @@ export async function updateArtifacts({
     ];
   }
 }
+
+async function getGoConstraints(
+  goModFileName: string
+): Promise<string | undefined> {
+  const content = (await readLocalFile(goModFileName, 'utf8')) ?? null;
+  if (!content) {
+    return undefined;
+  }
+  const re = regEx(/^go\s*(?<gover>\d+\.\d+)$/m);
```

Comment:
Are you sure about the `/m` modifier?

Reply:
I added the `m` flag in order to support the suggestion below:
https://github.com/renovatebot/renovate/pull/17178#discussion_r985503338
otherwise `^` and `$` will match the entire string (raw go mod).

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/multiline#description

I guess we can drop the `m` flag and use another means to achieve the same(split to lines and exec regex).

Reply:
no, this is ok.

Reply:
Sorry, that's right